plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -6785,6 +6785,7 @@ public Answer execute(DestroyCommand cmd) {
                 if (s_logger.isInfoEnabled()) {
                     s_logger.info("Destroy template volume " + vol.getPath());
                 }
+                vmMo.markAsVirtualMachine(hyperHost.getHyperHostOwnerResourcePool(), hyperHost.getMor());
```

Comment:
@Spaceman1984 why to mark as VirtualMachine before destroying it ? will the destroy operation fail without it?

Reply:
Yes, it did @sureshanaparti,

Reply:
I see delete/destory operation works for VM & Template in the vCenter. so, I think _destroyTask()_ API can be used with the template mor as well, no need to explicitly mark as VM. What are the errors seen in the logs for failed destroy task without marking as VM?

Reply:
If _destroy()_ works only when marked as a VM, then the same should be ensured across the usages of vmMO _destroy()_

Reply:
The error was:
Unable to delete template with snapshots - convert the template to a Virtual Machine first.

Reply:
@Spaceman1984 @sureshanaparti - in case of linked clones, it makes sense (the error). Do we/CloudStack check that template is not used by any of the VMs, prior to removing it. In that case it perfectly makes sense, if we don't delete the template from primary storage. Thoughts?

Reply:
Yes, in this case, better not to mark the VM as template while spooling it from the secondary storage. So, no need to mark it back to VM for deletion.

Reply:
The error happened for all templates I tested, even systemvm. I speculate that it is referring to the internal command that vmware calls when you click on the delete button from the UI.

Reply:
> Do we/CloudStack check that template is not used by any of the VMs, prior to removing it. 

Yes

Reply:
Looks good. I tried `vmMo.removeAllSnapshots();` before the destroy but fails. I wonder if templates GC worked before not being marked as templates.